project/packages/dtos/src/questions.ts

```diff
@@ -1,30 +1,20 @@
 import { z } from "zod";
 
-export enum Complexity {
-  Easy = "Easy",
-  Medium = "Medium",
-  Hard = "Hard",
-}
-
-export enum Category {
-  Strings = "Strings",
-  Algorithms = "Algorithms",
-  DataStructures = "Data Structures",
-  BitManipulation = "Bit Manipulation",
-  Recursion = "Recursion",
-  Databases = "Databases",
-  BrainTeaser = "Brain Teaser",
-  Arrays = "Arrays",
-}
-
-const ComplexitySchema = z.nativeEnum(Complexity);
-const CategorySchema = z.nativeEnum(Category);
+export const categorySchema = z.string().min(1);
+export const complexitySchema = z.string().min(1);
+
+export const getQuestionsQuerySchema = z.object({
+  title: z.string().optional(),
+  category: categorySchema.optional(),
+  complexity: complexitySchema.optional(),
+  includeDeleted: z.coerce.boolean().optional(),
+});
 
 const commonQuestionFields = z.object({
   q_title: z.string().min(1),
   q_desc: z.string().min(1),
-  q_category: z.array(CategorySchema),
-  q_complexity: ComplexitySchema,
+  q_category: z.array(categorySchema).min(1),
```

Comment:
should we add a check here for uniqueness in the array too? idt zod has a built in function for uniqueness currently, but can use this alternative suggested [here](https://github.com/colinhacks/zod/discussions/2316#discussioncomment-10728482)

Reply:
ah okay that seems good, i'll add it in.